contracts/mixnet/src/helpers.rs

```diff
@@ -65,9 +72,88 @@ pub(crate) fn scale_reward_by_uptime(
     Ok(uint128_to_decimal(scaled))
 }
 
+// Extracts the node identity and owner of a delegation from the bytes used as
+// key in the delegation buckets.
+fn extract_identity_and_owner(bytes: Vec<u8>) -> StdResult<(Addr, IdentityKey)> {
+    if bytes.len() < NAMESPACE_LENGTH {
+        return Err(StdError::parse_err(
+            "mixnet_contract::types::IdentityKey",
+            "Invalid type",
+        ));
+    }
+    let identity_size = u16::from_be_bytes([bytes[0], bytes[1]]) as usize;
+    let identity_bytes: Vec<u8> = bytes
+        .iter()
+        .skip(NAMESPACE_LENGTH)
+        .take(identity_size)
+        .copied()
+        .collect();
+    let identity = IdentityKey::from_utf8(identity_bytes)
+        .map_err(|_| StdError::parse_err("mixnet_contract::types::IdentityKey", "Invalid type"))?;
+    let owner_bytes: Vec<u8> = bytes
+        .iter()
+        .skip(NAMESPACE_LENGTH + identity_size)
+        .copied()
+        .collect();
+    let owner = Addr::unchecked(
+        String::from_utf8(owner_bytes)
+            .map_err(|_| StdError::parse_err("cosmwasm_std::addresses::Addr", "Invalid type"))?,
+    );
+
+    Ok((owner, identity))
+}
+
+// currently not used outside tests
+#[cfg(test)]
+// Converts the node identity and owner of a delegation into the bytes used as
+// key in the delegation buckets.
+pub(crate) fn identity_and_owner_to_bytes(identity: &IdentityKey, owner: &Addr) -> Vec<u8> {
+    let mut bytes = u16::to_be_bytes(identity.len() as u16).to_vec();
+    bytes.append(&mut identity.as_bytes().to_vec());
+    bytes.append(&mut owner.as_bytes().to_vec());
+
+    bytes
+}
+
+pub(crate) fn get_all_delegations_paged<T>(
+    bucket: &ReadonlyBucket<T>,
+    start_after: &mut Option<Vec<u8>>,
```

Comment:
seeing this as mutable is a big red flag...

Reply:
Maybe I went a little bit to C-like here :D
I don't really see it to be very dangerous, but I will change it to go into the returned value, wrapped around in a structure, if you'd like